### PR TITLE
New readme

### PR DESCRIPTION
**Pull request by tonycurtis**
## What
Updates the UCX README with much more information

## Why ?
README didn't have much in it - want to add URLs, badges, licensing, ...

### CODE REVIEW

**Comment by shamisp** [comment]
@miked-mellanox @yosefe Please take a look. This is proper readme.md for the project.

**Comment by shamisp** [comment]
@bbenton

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8904/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@tonycurtis can you please add link with preview of the readme (a.k.a your branch)

**Comment by shamisp** [comment]
Okay, found the tree https://github.com/tonycurtis/ucx/tree/new-readme

File: README [hunk 1]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
+| Component | Role        | Description |
+| :---:     | :---:       | ---         |
+| UCP | Protocol          | Implements high-level abstractions such as tag-matching, streams, connection negotiation and establishment, multi-rail, and handling different memory types |
+| UCT | Transport         | Implements low-level communication primitives such as active messages, remote memory access, and atomic operations |
+| UCS | Services          | A collection of data structures, algorithms, and system utilities for common use |
+| UCM | Memory            | Intercepts memory allocation and release events, used by the  memory registration cache |
+
+## Supported Transports
+
+* [Infiniband](https://www.infinibandta.org/)
```
Comment (yosefe) [inline README]: let's add OPA

File: README [hunk 2]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
+* [RoCE](http://www.roceinitiative.org/)
+* [Cray Gemini and Aries](https://www.cray.com/)
+* [CUDA](https://developer.nvidia.com/cuda-zone)
+* [ROCm](https://rocm.github.io/)
+* Shared Memory
+    * posix, sysv, cma, knem, and xpmem
```
Comment (yosefe) [inline README]: we can add links to knem, xpmem, cma

File: README [hunk 3]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
+* TCP/IP fallback
```
Comment (yosefe) [inline README]: maybe just TCP/IP, without "fallback" word?

File: README [hunk 4]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
+<img src="doc/doxygen/Architecture.png">
+
+| Component | Role        | Description |
+| :---:     | :---:       | ---         |
+| UCP | Protocol          | Implements high-level abstractions such as tag-matching, streams, connection negotiation and establishment, multi-rail, and handling different memory types |
+| UCT | Transport         | Implements low-level communication primitives such as active messages, remote memory access, and atomic operations |
+| UCS | Services          | A collection of data structures, algorithms, and system utilities for common use |
+| UCM | Memory            | Intercepts memory allocation and release events, used by the  memory registration cache |
+
+## Supported Transports
```
Comment (yosefe) [inline README]: How about supported CPU architectures?

File: README [hunk 5]
```diff
@@ -1,30 +1,53 @@
...
+
+# Unified Communication X
+
+Unified Communication X (UCX) provides an optimized communication
+layer for Message Passing ([MPI](https://www.mpi-forum.org/)),
+[PGAS](http://www.pgas.org/)/[OpenSHMEM](http://www.openshmem.org/)
+libraries and RPC/data-centric applications.
+
+UCX utilizes high-speed networks for inter-node communication, and
+shared memory mechanisms. for efficient intra-node communication.
```
Comment (miked-mellanox) [inline README]: redundant dot in the middle of sentence ```memory mechanisms. for efficient ```

File: README (hunk 3 above)
Comment (shamisp) [reply README]: Our TCP support is first class citizen

File: README (hunk 4 above)
Comment (shamisp) [reply README]: Good point. Let's add x86_64, Power8/9, Arm v8

File: README [hunk 6]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
+* [Github](http://www.github.com/openucx/ucx/)
+* [Software Releases](http://www.github.com/openucx/ucx/releases)
+* [Mailing List](https://elist.ornl.gov/mailman/listinfo/ucx-group)
+
+## Licenses
+
+UCX comes with 2 licenses:
+
+* [BSD3](LICENSE)
+* [Apache 2.0](LICENSES.apache20)
```
Comment (shamisp) [inline README]: Our license is BSD3 only. I think we have only one file is re-distributed under Apache.
@yosefe I think we have explicitly list that only one particular file is re-distributed under apache.

File: README [hunk 7]
```diff
@@ -35,14 +58,72 @@ Connect client:
...
-- UCS - UCX Services. A collection of data structures, algorithms, and system
-  utilities for common use.
-- UCM - UCX Memory. Intercepts memory allocation and release events, used by
-  memory registration cache.
+## Our Community
+
+* [Project Website](http://www.openucx.org/)
+* [Github](http://www.github.com/openucx/ucx/)
+* [Software Releases](http://www.github.com/openucx/ucx/releases)
+* [Mailing List](https://elist.ornl.gov/mailman/listinfo/ucx-group)
```
Comment (shamisp) [inline README]: Maybe add link to twitter https://twitter.com/openucx

Comment (shamisp) [reply README]: There is actually badge for this https://github.com/badges/shields

**Comment by tonycurtis** [comment]
Added suggestions/fixes

**Comment by shamisp** [comment]
Looks good to me. @yosefe @miked-mellanox

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8936/ for details (Mellanox internal link).

**Comment by bbenton** [comment]
FWIW, 👍 from me.

File: README (hunk 6 above)
Comment (shamisp) [reply README]: there is PR to remove Apache file from the repo. So, let't go ahead and remove the reference to Apache License.

**Comment by shamisp** [comment]
Okay, I will go ahead and merge. If there are any concerns we can solve it through new PR.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8942/ for details (Mellanox internal link).

**Pull request merged**